src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -43,26 +47,149 @@ struct segment_info_t {
   }
 };
 
+// position within segment_info_set_t
+using segment_pos_t = uint64_t;
```

Comment:
Is this meant to avoid a map lookup?  I don't think it's worth the complexity, device_segment_t (rename to segment_id_t, see my other comment) already uniquely identifies an entry.

Reply:
> Is this meant to avoid a map lookup? I don't think it's worth the complexity, device_segment_t (rename to segment_id_t, see my other comment) already uniquely identifies an entry.

Yes, I thought with multiple devices as backends, the number of segments might be relatively large (32768 segments for a 4TB disk with 128MB segments), maybe I should group segments by devices. I mean somethine like `std::map<device_id_t, std::vector<segment_id_t>> segments`, what do you think about it?

Reply:
Why would that require a separate, external segment_pos_t type?

Reply:
> Why would that require a separate, external segment_pos_t type?

I thought segment_id_t may not be an appropriate name for the SegmentCleaner::segments index.

Reply:
segment_info_set_t maps a segment_id_t (device_segment_t at the moment) to a segment_info_t.  Internally, you are using a std::vector to actually store that mapping with an ancillary segment_managers map to store the device_id_t->offset mapping.  I'm fine with internal (private) methods of segment_info_set_t using a segment_pos_t type for indexing within that vector.  My objection is that the rest of SegmentCleaner also uses that internal type rather than simply using the segment_id_t operator[].